### ISSUE TITLE
Constants with type declarations

### ISSUE THREAD

**Issue by HertzDevil** [crystal-lang/crystal#13443]
This is not allowed:

```crystal
FOO : Int64 = 123
```

I think it should, because this allows constant initializers to use autocasting. In this case `FOO = 0_i64` would also suffice, but things change when the integer type does not have a fixed size, e.g. it is platform-specific:

```crystal
FOO : LibC::SizeT = 123
```

It is better than the following:

```crystal
FOO = LibC::SizeT.new(123)
```

because it is easier for the compiler to infer in the first case that `FOO` is a compile-time constant and skip the generation of the run-time initializer (`~FOO:init` [here](https://godbolt.org/z/M6z8nsYxv)).

**Comment by nobodywasishere** [comment]
Is there anything that would prohibit this from being implemented? I think this could be really useful, and would be happy to try implementing this.

**Comment by straight-shoota** [comment]
No, I don't think there are any obstacles. Just needs someone to pick it up. You're welcome to =)
